module/observations/helper.js

```diff
@@ -1784,9 +1788,26 @@ module.exports = class ObservationsHelper {
                             "allowMultipleAssessemts",
                             "license"
                     ]);
-                    
+
+                    const query = { 
+                        userId: userId,
+                        programId: observationData[0].programId,
+                    }
+        
+                    //Check data present in programUsers collection.
+                    programUsers = await programUsersHelper.programUsersDocuments(
```

Comment:
@VISHNUDAS-tunerlabs There can be a function in programUsersHelper that return true or false if the user has joined a program or not.